sdk/src/main/java/com/robotemi/sdk/Robot.kt

```diff
@@ -3267,6 +3319,83 @@ class Robot private constructor(private val context: Context) {
         return mapDataModel
     }
 
+    /**
+     * Get map elements, such as locations, virtual walls, green paths, etc.
+     *
+     * @return Map elements.
+     */
+    fun getMapElements(): List<Layer>? {
+//        if (isMapLocked() == true) return sdkService?.getMapElements(applicationInfo.packageName)
+        var cursor: Cursor? = null
+        val uriStr = StringBuffer("content://")
+            .append(SdkConstants.PROVIDER_AUTHORITY)
+            .append("/").append(SdkConstants.PROVIDER_PARAMETER_MAP_DATA)
+            .toString()
+        cursor = context.contentResolver.query(
+            Uri.parse(uriStr),
+            null,
+            null,
+            null,
+            null
+        )
+        if (cursor == null || !cursor.moveToFirst()) {
+            return null
+        }
+        val mapElementsJson = cursor.getString(cursor.getColumnIndexOrThrow(MAP_ELEMENTS))
+        cursor.close()
+        return gson.fromJson<List<Layer>>(
+            mapElementsJson,
+            object : TypeToken<List<Layer>>() {}.type
+        )
+    }
+
+    /**
+     * Get map image
+     *
+     * @return Map image
+     */
+    fun getMapImage(): MapImage? {
+        val gson = Gson()
+        val inputStream =
+            getInputStreamByMediaKey(ContentType.MAP_DATA_IMAGE, "") ?: return null
+        val inputStreamReader = InputStreamReader(inputStream)
+        var mapDataModel: MapDataModel? = null
+        try {
+            mapDataModel = gson.fromJson(
+                inputStreamReader,
+                MapDataModel::class.java
+            )
+        } catch (e: JsonParseException) {
+            Log.e(TAG, "getMapImage() - JSON parse error: ${e.message}")
+        } finally {
+            try {
+                inputStream.close()
+                inputStreamReader.close()
+            } catch (e: IOException) {
+                Log.e(TAG, "getMapImage() - ${e.message}")
+            }
+        }
+        return mapDataModel?.mapImage
+    }
+
+    /**
+     * Check if the map is lost.
+     *
+     * @return true if the map is lost, false otherwise.
+     */
+    fun isMapLost(): Boolean? {
+        return sdkService?.isMapLost
+    }
+
+    /**
+     * Check if the map is locked.
+     *
+     * @return true if the map is locked, false otherwise.
+     */
+    fun isMapLocked(): Boolean? {
+        return sdkService?.isMapLocked
```

Comment:
1. add try catch for RemoteException
2. aidl return false for boolean for APIs not supported by remote/server side. Consider to use int in aidl